tests/load/locustfiles/locustfile.py

```diff
@@ -31,119 +48,392 @@ def _(parser: Any):
+
+    def on_ws_error(self, ws: WebSocketApp, error: Exception) -> None:
+        """Called when there is a WebSocketApp error or if an exception is raised in a WebSocket
+        callback function.
+
+        Args:
+            ws: WebSocket class object
+            error: Exception object
+        """
+        logger.error(f"Error: {str(error)}")
+
+        if isinstance(error, WebSocketConnectionClosedException):
+            # WebSocket closures are expected. Reconnect and don't send the exception to the UI.
+            self.ws = None  # WebSocket is not available for task use
+            self.connect()
+            return
+
+        self.environment.events.user_error.fire(
+            user_instance=self.context(), exception=error, tb=error.__traceback__
+        )
+
+    def on_ws_close(
+        self, ws: WebSocketApp, close_status_code: int | None, close_msg: str | None
+    ) -> None:
+        """Called when closing a websocket.
+
+        Args:
+            ws: WebSocket class object
+            close_status_code: WebSocket close status
+            close_msg: WebSocket close message
+        """
+        if close_status_code or close_msg:
+            logger.info(f"WebSocket closed. status={close_status_code} msg={close_msg}")
+        self.ws = None  # WebSocket is not available for task use
+
+    @task(weight=95)
+    def send_notification(self):
+        """Sends a notification to a registered endpoint while connected to Autopush."""
+        if not self.ws or not self.channels:
+            logger.info("Task 'send_notification' skipped.")
+            return
 
-    def _time_event(self, name: str) -> TimeEvent:
-        return TimeEvent(self, name)
+        endpoint_url: str = random.choice(list(self.channels.values()))
+        self.post_notification(endpoint_url)
 
     def connect(self) -> None:
-        self.ws = create_connection(
+        """Creates the WebSocketApp that will run indefinitely."""
+        ws = websocket.WebSocketApp(
             self.environment.parsed_options.websocket_url,
-            header={"Origin": "http://localhost:1337"},
-            ssl=False,
+            header=self.WEBSOCKET_HEADERS,
+            on_message=self.on_ws_message,
+            on_error=self.on_ws_error,
+            on_close=self.on_ws_close,
+            on_open=self.on_ws_open,
         )
+        ws.run_forever()
 
-    def disconnect(self) -> None:
-        self.ws.close()
+    def post_notification(self, endpoint_url: str) -> None:
+        """Send a notification to Autopush.
 
-    def hello(self) -> None:
+        Args:
+            endpoint_url: A channel destination endpoint url
+        Raises:
+            ZeroStatusRequestError: In the event that Locust experiences a network issue while
+                                    sending a notification.
         """
-        Send a 'hello' message to Autopush.
+        message_type: str = "notification"
+        data: str = "TestData" + "".join(
```

Comment:
I think this was covered in the discussion, but leading off the data with `TestData` followed by random garbage allows you to easily differentiate the payload from the incoming base64 encoded data segment (which can  also look like random garbage).

Again, not sure if you might want to add some comment here noting that since it was a big point of confusion yesterday.

Reply:
Yes! I will add a comment.